### PR TITLE
Move to gogoproto

### PR DESCRIPTION
**Pull request by fabxc**
@stuartnelson3 @mxinden 

This moves our generated protobuf to gogoproto, which allows us to get rid of the awkward time types. I'd like to get this in before a release of the notification fixes.

### CODE REVIEW

File: scripts/genproto.sh [hunk 1]
```diff
@@ -1,2 +1,37 @@
-protoc --go_out=. nflog/nflogpb/nflog.proto
-protoc --go_out=. silence/silencepb/silence.proto
+
+#!/usr/bin/env bash
+#
+# Generate all etcd protobuf bindings.
```
Comment (mxinden) [inline scripts/genproto.sh]: Why do we reference `etcd` here?

Comment (fabxc) [inline scripts/genproto.sh]: Stolen script ;)

Comment (brian-brazil) [inline scripts/genproto.sh]: What's the license on this file?

Comment (brian-brazil) [inline scripts/genproto.sh]: The source repo is all Apache 2.0, so I'd just add a one-line note of where it was originally taken from and under what license.

Comment (fabxc) [inline scripts/genproto.sh]: Done.

File: scripts/genproto.sh [hunk 2]
```diff
@@ -1,2 +1,40 @@
-protoc --go_out=. nflog/nflogpb/nflog.proto
-protoc --go_out=. silence/silencepb/silence.proto
+#!/usr/bin/env bash
+#
+# Generate all protobuf bindings.
+# Run from repository root.
+#
+# Initial script take from etcd under the Apache 2.0 license
```
Comment (brian-brazil) [inline scripts/genproto.sh]: taken

**Pull request merged**